lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/S3StorageClient.java

```diff
@@ -34,137 +31,147 @@ public class S3StorageClient extends BaseStorageClient {
   private S3Client s3;
   private static final Logger log = LoggerFactory.getLogger(S3StorageClient.class);
 
-  public S3StorageClient(URI pathToStorage, String docIdPrefix, List<Pattern> excludes, List<Pattern> includes,
-      Map<String, Object> cloudOptions, Config fileOptions) {
-    super(pathToStorage, docIdPrefix, excludes, includes, cloudOptions, fileOptions);
+  public S3StorageClient(Config s3CloudOptions) {
+    super(s3CloudOptions);
   }
 
   @Override
-  public void init() throws ConnectorException {
+  protected void validateOptions(Config cloudOptions) {
+    if (!validOptions(cloudOptions)) {
+      throw new IllegalArgumentException("Missing '" + S3_ACCESS_KEY_ID + "' or '" + S3_SECRET_ACCESS_KEY + "' or '" + S3_REGION + "' in cloudOptions for S3StorageClient.");
```

Comment:
exception message still references cloudOptions but we took that out -- should update in accordance with the new config format